src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -963,9 +963,22 @@ int xcldev::xclTop(int argc, char *argv[])
 }
 
 const std::string dsaPath("/opt/xilinx/dsa/");
-const std::string xsaPath("/opt/xilinx/xsa/");
 const std::string xrtPath("/opt/xilinx/xrt/");
 
+inline std::string getXsaPath(const uint16_t &vendor)
+{
+    std::string vendorName;
```

Comment:
Being that a "vendor" value of zero should result in an empty string.   It would be better to check the parameters first and if non-zero continue with the flow.  For example:
```
if (vendor == 0)
    return std::string();
```

Doing this (along with the suggested changes below), helps make the code more readable.